packages/firestore/test/unit/specs/recovery_spec.test.ts

```diff
@@ -75,5 +75,38 @@ describeSpec(
           .expectEvents(query, {});
       }
     );
+
+    specTest(
+      'Query is listened to by primary (with recovery)',
+      ['multi-client'],
+      () => {
+        const query = Query.atPath(path('collection'));
+
+        return (
+          client(0)
+            .expectPrimaryState(true)
+            .failDatabase()
+            .client(1)
+            .userListens(query)
+            .client(0)
+            // The primary client 0 receives a WebStorage notification about the
+            // new query, but it cannot load the target from IndexedDB. The
+            // query will only be listened to once we recover the database.
+            .recoverDatabase()
+            .runTimer(TimerId.AsyncQueueRetry)
+            .expectListen(query)
+            .failDatabase()
+            .client(1)
+            .userUnlistens(query)
+            .client(0)
+            // The primary client 0 receives a notification that the query can
+            // be released, but it can only process the change after we recover
+            // the database.
+            .recoverDatabase()
+            .runTimer(TimerId.AsyncQueueRetry)
+            .expectActiveTargets()
```

Comment:
Should we expect that there is an active target before recoverDatabase?

Reply:
The targets are checked at each step, but adding this here explicitly makes it much clearer what is going on. Done.